extensions/layout/init.lua

```diff
@@ -10,6 +10,7 @@ local fnutils = require("hs.fnutils")
 local screen = require("hs.screen")
 local window = require("hs.window")
 local application = require("hs.application")
+local spi = require("hs._asm.undocumented.spaces.internal")
```

Comment:
The `hs._asm.undocumented.spaces.internal` module isn't shipped with Hammerspoon, so we can't include it in one of our modules.

Reply:
Maybe I can revise so that caller should pass in an actually screen previously determined by `hs._asm.undocumented.spaces.internal.UUIDforScreen` that can be vetted by the existing code?  I.e. no change really.

In other notes, I have a preprocessing enhancement for use of `hs.layout` that can deal with multi-monitors with layout specified like the example form below.  Perhaps I should submit a separate issue to discuss possibly incorporating this into hs.layout?

```
oneMonitorDef = {
  {
    screen = "Color LCD";
    display_uuid = "A1067B4B-926A-055F-C929-DD7875BE07A3";
  },
}

twoMonitorDef  = {
  {
    screen = "Color LCD";
    display_uuid = "A1067B4B-926A-055F-C929-DD7875BE07A3";
  },
  {
    screen = "DELL U3011";
    display_uuid = "61346AEB-0959-1BE6-A8BE-36B9E98B97FC";
  },
}

windows = {
  {
    app = "Atom",
    title = ".*~/lsrc/data.and.reviews"
    oneMonitor = {
      geom = geoms.right75,
      screen_num = 1,
      space = 3
    },
    twoMonitor = {
      geom = geoms.right50,
      screen_num = 2,
      space = 1
    },
  },
...
}
```

Reply:
FWIW it looks like the UUID is officially moving to being private API in 10.14. I'm also not convinced that it's properly unique - e.g. is the UUID stable across connections of the monitor?

My overall approach for hs.layout has always been to identify the monitors ahead of time - e.g. with two identical monitors I'd get `hs.screen.primaryScreen()` and then use `:toEast()` and `:toWest()` to determine which monitor is which, and pass them to hs.layout as `leftMonitor` and `rightMonitor`.